protocol/x/govplus/keeper/msg_server.go

```diff
@@ -15,3 +21,40 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 }
 
 var _ types.MsgServer = msgServer{}
+
+func (k msgServer) SlashValidator(
+	goCtx context.Context,
+	msg *types.MsgSlashValidator,
+) (*types.MsgSlashValidatorResponse, error) {
+	if !k.HasAuthority(msg.Authority) {
+		return nil, errorsmod.Wrapf(
+			govtypes.ErrInvalidSigner,
+			"invalid authority %s",
+			msg.Authority,
+		)
+	}
+
+	consAddr, err := sdk.ConsAddressFromBech32(msg.ValidatorAddress)
+	if err != nil {
+		return nil, types.ErrValidatorAddress
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	_, err = k.stakingKeeper.Slash(
```

Comment:
curious - why do we call `stakingKeeper.Slash` here and not `slashingKeeper.Slash`? slashing calls staking.Slash but also emits some events

Reply:
No particular reason. We didn't have any use for that event.